src/commands/auth/login.ts

```diff
@@ -6,17 +6,51 @@ import { askCredentials } from "./util";
 import { cliContainer } from "../../container";
 
 async function run(argv: Arguments) {
-  const answers = await askCredentials(argv);
+  let password: string | undefined;
+  if (argv["password-stdin"]) {
+    password = await readStdin();
+    password = password.replace(/\r\n$|\n$|\r$|\n\r$/g, "");
+  }
+  const answers = await askCredentials({
+    email: argv.email as string,
+    password: password,
+  });
   await cliContainer.container.auth.login(answers.email, answers.password);
   console.log(chalk`Login as {green ${answers.email}}.`);
 }
 
+async function readStdin(): Promise<string> {
+  return new Promise((resolve, reject) => {
+    let data = "";
+    const stdin = process.stdin;
+
+    stdin.on("readable", function() {
```

Comment:
Why `readable` is used? It seems when I type `<password>^D` in shell, it would just terminate instead of using the provided password.

Reply:
Changed to use the `data` event instead!